### PR TITLE
fix less bug for #472

### PR DESCRIPTION
**Pull request by kaysonwu**
fix less bug for [#472](https://github.com/kristerkari/stylelint-scss/issues/472)

### CODE REVIEW

**Comment by kaysonwu** [comment]
@kristerkari

File: src/rules/selector-no-redundant-nesting-selector/README.md [hunk 1]
```diff
@@ -59,3 +59,51 @@ p {
...
 ```
+
+## Options
+
+Some false alarms can be blocked by assigning keywords:
+
+```js
+{
+  rules: {
+    'scss/selector-no-redundant-nesting-selector', [true, { keywords: ['when'] }],
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/README.md:70]: The standard way to name the option would be to call it `ignoreKeywords`. But that makes me think that is `keyword` the correct term for this? Is that what it's called in Less?

File: src/rules/selector-no-redundant-nesting-selector/README.md [hunk 2]
```diff
@@ -59,3 +59,51 @@ p {
   }
 }
 ```
+
+## Options
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/README.md:63]: This should follow the formatting for options that is used in `stylelint` and other `stylelint-scss` rules, e.g. https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/at-function-named-arguments/README.md#options

File: src/rules/selector-no-redundant-nesting-selector/README.md (hunk 1 above)
Comment (kaysonwu) [reply src/rules/selector-no-redundant-nesting-selector/README.md:70]: I took it to mean reserved. I can follow your principle if you think it necessary

File: src/rules/selector-no-redundant-nesting-selector/README.md [hunk 3]
```diff
@@ -59,3 +59,51 @@ p {
...
+
+Some false alarms can be blocked by assigning keywords:
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/README.md:65]: This is not a good way of describing the option. 

A better way would be to say something like: if you are using Less or some other non-SCSS syntax, the warnings can be disabled by using this option.

File: src/rules/selector-no-redundant-nesting-selector/README.md [hunk 4]
```diff
@@ -59,3 +59,51 @@ p {
...
+
+```js
+{
+  rules: {
+    'scss/selector-no-redundant-nesting-selector', [true, { keywords: ['when'] }],
+  },
+}
+```
+
+The following patterns are *not* considered warnings:
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/README.md:75]: There could be some kind of additional text here to say that this is Less syntax.

File: src/rules/selector-no-redundant-nesting-selector/__tests__/index.js [hunk 5]
```diff
@@ -345,6 +345,50 @@ testRule(rule, {
...
+    },
+    {
+      code: `
+      @theme: ~'dark';
+      p {
+        & when (@theme = dark) {}
+        & when not (@theme = dark) {}
+      }
+    `,
+      line: 4,
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/__tests__/index.js:357]: Also add the `column` property here.

File: src/rules/selector-no-redundant-nesting-selector/README.md (hunk 1 above)
Comment (kristerkari) [reply src/rules/selector-no-redundant-nesting-selector/README.md:70]: Please have a look at this guide. It explains how to name options in stylelint rules:
https://stylelint.io/user-guide/rules/about#options

**Comment by kaysonwu** [comment]
I've received your suggestion. I'm dealing with it

**Comment by kaysonwu** [comment]
@kristerkari You need to review.

File: src/rules/selector-no-redundant-nesting-selector/__tests__/index.js (hunk 5 above)
Comment (kristerkari) [reply src/rules/selector-no-redundant-nesting-selector/__tests__/index.js:361]: @kaysonwu can you still fix this one? The `column` property is missing from the test.

Comment (kaysonwu) [reply src/rules/selector-no-redundant-nesting-selector/__tests__/index.js:361]: I've fixed it, but it's not synced. I'll check it

Comment (kaysonwu) [reply src/rules/selector-no-redundant-nesting-selector/__tests__/index.js:361]: Now, it seems to be synchronized, I added the `color` property

File: src/rules/selector-no-redundant-nesting-selector/index.js [hunk 6]
```diff
@@ -1,27 +1,34 @@
...
 export const ruleName = namespace("selector-no-redundant-nesting-selector");
 
 export const messages = utils.ruleMessages(ruleName, {
   rejected: "Unnecessary nesting selector (&)"
 });
 
-export default function(actual) {
+export default function(actual, options) {
   return (root, result) => {
     const validOptions = utils.validateOptions(result, ruleName, { actual });
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/index.js:17]: Oh sorry I totally missed it on the first review round, but the `utils.validateOptions` call should include rule's secondary options too.

Have a look at how it's done here:
https://github.com/stylelint/stylelint/blob/master/lib/rules/font-family-no-duplicate-names/index.js#L25-L36

The option could also accept a regex as a value like the rule I just linked.

File: src/rules/selector-no-redundant-nesting-selector/README.md [hunk 7]
```diff
@@ -59,3 +59,57 @@ p {
   }
 }
 ```
+
+## Options
+
+`string[]`: `[keywords]`
```
Comment (kristerkari) [inline src/rules/selector-no-redundant-nesting-selector/README.md:65]: Sorry for nitpicking, but this should be `ignoreKeywords: ["/regex/", /regex/, "string"]`. The rule could be updated to also support regex values.

Here's an example:
https://github.com/stylelint/stylelint/tree/master/lib/rules/font-family-no-duplicate-names#ignorefontfamilynames-regex-regex-string

Comment (kaysonwu) [reply src/rules/selector-no-redundant-nesting-selector/README.md:65]: This is the right pick，I will update immediately, remember to wait for me

**Comment by kaysonwu** [comment]
@kristerkari   Please review it again, I am waiting for the new version

**Comment by kaysonwu** [comment]
@kristerkari When can you review my PR?

**Comment by kaysonwu** [comment]
Never mind, is there a plan to release a new version?

**Comment by kristerkari** [comment]
> Never mind, is there a plan to release a new version?

Yes, I'm just going to quickly see if I can include one more bug fix for the release. I'll release the new version asap.

**Pull request merged**